src/Common/ErrorCodes.cpp

```diff
@@ -610,6 +611,23 @@ namespace ErrorCodes
     constexpr ErrorCode END = 1002;
     ErrorPairHolder values[END + 1]{};
 
+    std::set retryable_errors = {
+        3,   // UNEXPECTED_END_OF_FILE
+        159, // TIMEOUT_EXCEEDED
+        164, // READONLY
+        202, // TOO_MANY_SIMULTANEOUS_QUERIES
+        203, // NO_FREE_CONNECTION
+        209, // SOCKET_TIMEOUT
+        210, // NETWORK_ERROR
+        242, // TABLE_IS_READ_ONLY
```

Comment:
TABLE_IS_READ_ONLY may be retriable or not, it depends

Reply:
Is it possible to introduce a new error code for retryable TABLE_IS_READ_ONLY scenarios?